### PR TITLE
Various analytics service provider additions

### PR DESCRIPTION
**Pull request by thehouseplant**
Just wanted to put in a PR to cover some more analytics providers to the mix. Most of the ones I have added have generous free tiers, so it seems like a worthwhile addition.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -168,12 +168,18 @@ There is a single parameter object with sensible defaults that can be overridden
 ```yaml
 paige:
   analytics:
+    beam: # See https://beamanalytics.io/
```
Comment (willfaught) [inline README.md:171]: Drop the trailing /

File: README.md [hunk 2]
```diff
@@ -168,12 +168,18 @@ There is a single parameter object with sensible defaults that can be overridden
...
+      token: ""
     chartbeat: # See https://chartbeat.com
       account_id: ""
       domain: ""
     disabled: false # Disable all analytics
+    clarity: # See https://clarity.microsoft.com/
```
Comment (willfaught) [inline README.md:177]: Drop the trailing /

File: README.md [hunk 3]
```diff
@@ -168,12 +168,18 @@ There is a single parameter object with sensible defaults that can be overridden
...
+      account_id: ""
     clicky: # See https://clicky.com
       account_id: ""
+    counter: # See https://counter.dev/
```
Comment (willfaught) [inline README.md:181]: Drop the trailing /

File: README.md [hunk 4]
```diff
@@ -186,6 +192,8 @@ paige:
       host_url: ""
     mixpanel: # See https://mixpanel.com
       token: ""
+    pirsch: # See https://pirsch.io/
```
Comment (willfaught) [inline README.md:195]: Drop the trailing /

File: layouts/partials/paige/analytics/beam.html
Comment (willfaught) [inline layouts/partials/paige/analytics/beam.html:1]: Needs a trailing line feed

File: layouts/partials/paige/analytics/clarity.html
Comment (willfaught) [inline layouts/partials/paige/analytics/clarity.html:1]: Needs a trailing line feed

File: layouts/partials/paige/analytics/counter.html
Comment (willfaught) [inline layouts/partials/paige/analytics/counter.html:1]: Needs a trailing line feed

File: layouts/partials/paige/analytics/pirsch.html
Comment (willfaught) [inline layouts/partials/paige/analytics/pirsch.html:1]: Needs a trailing line feed

File: layouts/partials/paige/analytics/clarity.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{{ $page := . }}
+
+{{ with $page.Param "paige.analytics.clarity" }}
+    <script type="text/javascript">
+        (function(c,l,a,r,i,t,y){
+            c[a]=c[a]||function(){(c[a].q=c[a].q||[]).push(arguments)};
+            t=l.createElement(r);t.async=1;t.src="https://www.clarity.ms/tag/"+i;
+            y=l.getElementsByTagName(r)[0];y.parentNode.insertBefore(t,y);
+        })(window, document, "clarity", "script", "{{ .account_id }}");
```
Comment (willfaught) [inline layouts/partials/paige/analytics/clarity.html:9]: We need to mark `.account_id` as safe:

```js
"{{ .account_id | safeJS }}"
```

File: layouts/partials/paige/analytics/counter.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{{ $page := . }}
+
+{{ with $page.Param "paige.analytics.counter" }}
+    <script src="https://cdn.counter.dev/script.js" data-id="{{ .account_id }}" data-utcoffset="-7"></script>
```
Comment (willfaught) [inline layouts/partials/paige/analytics/counter.html:4]: Do we need to parameterize utcoffset?

Is this meant to be the user's default time zone offset? If so, what happens if the browser is in another time zone?

**Review by willfaught — CHANGES_REQUESTED** [review]
Excellent, I hadn't heard of these!

**Comment by willfaught** [comment]
Thanks for the changes. Two comments remain.

File: layouts/partials/paige/analytics/counter.html (hunk 6 above)
Comment (thehouseplant) [reply layouts/partials/paige/analytics/counter.html:4]: This is a good question. I'll read the documentation more before I submit an update for this.

File: README.md [hunk 7]
```diff
@@ -180,6 +180,7 @@ paige:
       account_id: ""
     counter: # See https://counter.dev
       account_id: ""
+      utc_offset: "" # GMT +/- local timezone
```
Comment (willfaught) [inline README.md:183]: GMT is different than UTC (https://en.wikipedia.org/wiki/UTC_offset). We should also document that it should be just the hour integer without a leading zero and no minutes.

Perhaps something like `# UTC offset; examples are "-7", "0", and "7"`?

**Review by willfaught — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (willfaught) [reply README.md:183]: (I'm assuming that "0" and "7" are correct.)

**Comment by willfaught** [comment]
Going to merge as-is so I can convert the readme YAML to TOML. Thanks again!

**Pull request merged**